Microsoft.NET.Build.Containers.IntegrationTests/DockerSupportsArchInlineData.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Text.Json;
+using Microsoft.DotNet.CommandUtils;
+using Xunit.Sdk;
+
+namespace Microsoft.NET.Build.Containers.IntegrationTests;
+public class DockerSupportsArchInlineData : DataAttribute
+{
+    // tiny optimization - since there are many instances of this attribute we should only get
+    // the daemon status once
+    private static Task<bool> IsDaemonAvailable = new LocalDocker(Console.WriteLine).IsAvailable();
+
+    // also an optimization - this doesn't change over time so we can compute it once
+    private static string[] LinuxPlatforms = GetSupportedLinuxPlatforms();
+
+    // another optimization - daemons don't switch types easily or quickly, so this is as good as static
+    private static Task<bool> IsWindowsDaemon =
+        IsDaemonAvailable.ContinueWith(t => t.Result && GetIsWindowsDaemon());
+
+    private readonly string _arch;
+    private readonly object[] _data;
+
+    public DockerSupportsArchInlineData(string arch, params object[] data)
+    {
+        _arch = arch;
+        _data = data;
+    }
+
+    public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+    {
+        if (DaemonSupportsArch(_arch))
+        {
+            var totalArray = new object[_data.Length + 1];
+            totalArray[0] = _arch;
+            Array.Copy(_data, 0, totalArray, 1, _data.Length);
+            return new[] { totalArray };
+        };
+        return Array.Empty<object[]>();
+    }
+
+    private bool DaemonSupportsArch(string quemu_Arch)
+    {
+        if (IsDaemonAvailable.GetAwaiter().GetResult())
```

Comment:
we use this method several time in blocking context, it might reasonable to create a sync call for that. 
at first glance, i see no impediments for that.